pkg/apis/core/validation/shoot.go

```diff
@@ -1923,6 +1923,29 @@ func validateMaintenance(maintenance *core.Maintenance, fldPath *field.Path, wor
 		}
 	}
 
+	if maintenance.AutoRotation != nil && maintenance.AutoRotation.Credentials != nil {
+		credentials := maintenance.AutoRotation.Credentials
+		credentialsPath := fldPath.Child("autoRotation", "credentials")
+
+		if credentials.Observability != nil && credentials.Observability.RotationPeriod != nil && credentials.Observability.RotationPeriod.Duration != 0 &&
+			(credentials.Observability.RotationPeriod.Duration < 30*time.Minute || credentials.Observability.RotationPeriod.Duration > 90*24*time.Hour) {
```

Comment:
Maintenance is performed only once per day, it doesn't make too much sense to have a rotation period of less than 24 hours as it might be confusing.